cardano-cli/test/Test/Golden/Shelley/Governance/AnswerPoll.hs

```diff
@@ -0,0 +1,145 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Golden.Shelley.Governance.AnswerPoll
+  ( golden_shelleyGovernanceAnswerPollVrf
+  , golden_shelleyGovernanceAnswerPollCold
+  , golden_shelleyGovernanceAnswerPollInvalidAnswer
+  ) where
+
+import           Hedgehog (Property)
+import           Test.OptParse
+
+import qualified Data.List as L
+import qualified Hedgehog as H
+
+{- HLINT ignore "Use camelCase" -}
+
+golden_shelleyGovernanceAnswerPollVrf :: Property
+golden_shelleyGovernanceAnswerPollVrf = propertyOnce $ do
+  pollFile <- noteInputFile "test/data/golden/shelley/governance/poll.json"
+  vrfKeyFile <- noteInputFile "test/data/golden/shelley/governance/vrf.sk"
+
+  stdout <- execCardanoCLI
+    [ "governance", "answer-poll"
+    , "--poll-file", pollFile
+    , "--signing-key-file", vrfKeyFile
+    , "--answer", "0"
+    ]
+
+  H.assert $ json `L.isInfixOf` stdout
+ where
+  json = unlines
+    [ "{"
+    , "    \"94\": {"
+    , "        \"map\": ["
+    , "            {"
+    , "                \"k\": {"
+    , "                    \"int\": 2"
+    , "                },"
+    , "                \"v\": {"
+    , "                    \"bytes\": \"c9a077a1098d73498d17e9ea27045af820c311ced91f8c2bb9b5c7f446379063\""
+    , "                }"
+    , "            },"
+    , "            {"
+    , "                \"k\": {"
+    , "                    \"int\": 3"
+    , "                },"
+    , "                \"v\": {"
+    , "                    \"int\": 0"
+    , "                }"
+    , "            },"
+    , "            {"
+    , "                \"k\": {"
+    , "                    \"int\": 4"
+    , "                },"
+    , "                \"v\": {"
+    , "                    \"list\": ["
+    , "                        {"
+    , "                            \"bytes\": \"2dc2fa217af8b52251c4cdf538fa106cbf0b5beac3e74d05f97ceb33c0147a2c\""
+    , "                        },"
+    , "                        {"
+    , "                            \"list\": ["
+    , "                                {"
+    , "                                    \"bytes\": \"3606afd8d437a69a0c94ebff7fe57638bdb54bd03193c3c830808187a5d740222ab26fa3f1b7df8b22e846196dda1ee16d4d6e64c062129e07273ec36c06f453\""
+    , "                                },"
+    , "                                {"
+    , "                                    \"bytes\": \"06c46025c18deaa04aa45913e1c91906\""
+    , "                                }"
+    , "                            ]"
+    , "                        }"
+    , "                    ]"
+    , "                }"
+    , "            }"
+    , "        ]"
+    , "    }"
+    , "}"
+    ]
```

Comment:
Golden test data should go into files under `data/golden`, like `data/golden/shelley/governance`

Reply:
Yeah, golden tests are constructed so that expected outputs live in files and we diff actual output against the file.

Reply:
https://github.com/input-output-hk/cardano-node/compare/c018de7877eb6c513ddc029224ba2a189c4f6b2c..7adb11180e227089a5aa44845a06737efacf3401